notary-btc-integration-test/src/integration-test/kotlin/integration/btc/BtcAddressGenerationIntegrationTest.kt

```diff
@@ -51,7 +50,18 @@ class BtcAddressGenerationIntegrationTest {
         GlobalScope.launch {
             environment.btcAddressGenerationInitialization.init().failure { ex -> throw ex }
         }
-        Thread.sleep(WAIT_PREGEN_INIT_MILLIS)
+        // Wait for initial address generation
+        Thread.sleep(WAIT_PREGEN_PROCESS_MILLIS * environment.btcGenerationConfig.threshold)
```

Comment:
Can we move this part to the `environment` object in order to reduce duplications and test code size